### PR TITLE
Add an export function for KubeConfigs. Add unit tests too.

### PR DESCRIPTION
**Pull request by brendandburns**
Partial fix for #340

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -320,6 +330,20 @@ export class KubeConfig {
...
+            apiVersion: 'v1',
+            kind: 'Config',
+            clusters: this.clusters.map(exportCluster),
+            users: this.users.map(exportUser),
+            contexts: this.contexts.map(exportContext),
+            preferences: {},
+            'current-context': this.getCurrentContext(),
+        };
+
+        return JSON.stringify(configObj);
```
Comment (drubin) [inline src/config.ts]: Is there a reason we are exporting the data as json and not an object? Just wondering if people would want the json or maybe rather use yaml?

I didn't look at the go implementation though

Comment (brendandburns) [reply src/config.ts]: JSON is valid YAML :)

The purpose of this is to serialize it to a string. I can change the name to 'SerializeConfig' if that's clearer.

Comment (drubin) [reply src/config.ts]: Ah yes, I totally forgot. 

I was just thinking most people would use this to write their kubeconfig which is currently in `yaml` and if their intention was to convert it to `yaml` to be human-readable they would need to parse this and then convert it anyway.

**Comment by drubin** [comment]
/lgtm

We can address the other issues or name in another PR?

**Pull request merged**